src/utilities/shared-specs.ts

```diff
@@ -0,0 +1,34 @@
+// -----------------------------------------------------------------------------------------
+// #region Public Functions
+// -----------------------------------------------------------------------------------------
+
+/**
+ * Utility function for running a test body a certain number of times. Useful for ensuring specific
+ * behavior on implementations that return randomized data and edge cases will not immediately be
+ * exposed.
+ *
+ * @param {string} name Name of the test
+ * @param {Function} fn Function body to run (can be asynchronous)
+ * @param {number} [times=1000] Number of times to run the test function.
+ */
+const testLoop = (name: string, fn: Function, times: number = 1000) => {
```

Comment:
maybe consider lowering the default value for `times`; I feel like for most things, around 100 should be enough, and it would make the tests run faster.

Reply:
Fair enough 😛 I'll lower it and swap some implementations to make sure it's still catching these patch cases we have